pkg/models/datasource.go

```diff
@@ -81,6 +81,15 @@ func (ds DataSource) AllowedCookies() []string {
 	return []string{}
 }
 
+// Specific error type for grpc secrets management so that we can show more detailed plugin errors to users
+type ErrDatasourceSecretsPluginUserFriendly struct {
+	Err error
```

Comment:
Nit. Could be a string I guess?

Reply:
done [31f76a8](https://github.com/grafana/grafana/pull/50542/commits/31f76a88d85b19a0e1e263e771e581d22e1a1fb8)